package.json

```
@@ -46,7 +46,11 @@
     "build:immutable": "bunchee index.ts --cwd immutable -m --no-sourcemap",
     "prepublishOnly": "yarn clean && yarn build",
     "publish-beta": "yarn publish --tag beta",
-    "types:check": "tsc --noEmit",
+    "types:check": "yarn types:check:core && yarn types:check:infinite && yarn types:check:immutable && yarn types:check:fetcher",
+    "types:check:core": "tsc --noEmit",
+    "types:check:infinite": "cd infinite && tsc --noEmit",
+    "types:check:immutable": "cd immutable && tsc --noEmit",
+    "types:check:fetcher": "cd test-type && tsc --noEmit",
```

we can reduce these scripts by running type checking on top root dir to include all assets. I filed https://github.com/vercel/swr/pull/1492 to reduce them

```suggestion
    "types:check": "tsc --noEmit",
```

---

But it would create test-type/fetcher.d.ts in dist folder. Could it  be avoided ?